src/Uno.UI/UI/Xaml/UIElement.wasm.cs

```diff
@@ -571,37 +572,34 @@ internal virtual void ManagedOnUnloaded()
 		// We keep track of registered routed events to avoid registering the same one twice (mainly because RemoveHandler is not implemented)
 		private RoutedEventFlag _registeredRoutedEvents;
 
-		partial void AddFocusHandler(RoutedEvent routedEvent, int handlersCount, object handler, bool handledEventsToo)
+		/// <summary>
+		/// Initialize handler for native focus event.
+		/// </summary>
+		private void TrackFocus()
 		{
-			if (handlersCount != 1
-				// We do not remove event handlers for now, so do not rely only on the handlersCount and keep track of registered events
-				|| _registeredRoutedEvents.HasFlag(routedEvent.Flag))
-			{
-				return;
-			}
-			_registeredRoutedEvents |= routedEvent.Flag;
-
-			string domEventName;
-			if (routedEvent.Flag == RoutedEventFlag.GotFocus)
-			{
-				domEventName = "focus";
-			}
-			else
-			{
-				domEventName = routedEvent.Flag == RoutedEventFlag.LostFocus
-					? "focusout"
-					: throw new ArgumentOutOfRangeException(nameof(routedEvent), "Not a focus event");
-			}
+			var handler = this is Control ? (Func<bool>)ProcessControlFocused : ProcessElementFocused;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Offload the code that's conditional on this 'is' test to the appropriate subclass and remove the test.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5251853)